ios/storage.md

```diff
@@ -728,19 +607,38 @@ After adding the new mutation in our operations file, we run the code generator
 ```
 Update the `AWSAppSyncClientConfiguration` object to provide the `AWSS3TransferUtility` client for managing the uploads and downloads:
 ```swift
-  let appSyncConfig = try AWSAppSyncClientConfiguration(url: AppSyncEndpointURL,
+let appSyncConfig = try AWSAppSyncClientConfiguration(url: AppSyncEndpointURL,
                                                       serviceRegion: AppSyncRegion,
                                                       credentialsProvider: credentialsProvider,
                                                       databaseURL:databaseURL,
                                                       s3ObjectManager: AWSS3TransferUtility.default())
 ```
-The mutation operation doesn't require any specific changes in method signature. It requires only an S3ObjectInput with bucket, key, region, localUri, and mimeType. Now when you do a mutation, it automatically uploads the specified file to Amazon S3 using the `AWSS3TransferUtility` client internally.
 
+**The `AWSS3ObjectManager` integration** 
+
+In order to use `AWSS3TransferUtility` as the `s3ObjectManager` the consumer needs to provide an integration layer between the base types (`S3ObjectInput`, `AWSS3TransferUtility`) and the required protocols: `AWSS3ObjectProtocol`, `AWSS3InputObjectProtocol` and `AWSS3ObjectManager`.
+
+```swift
+extension S3ObjectInput: AWSS3ObjectProtocol, AWSS3InputObjectProtocol {
+    // ...
+}
+
+extension AWSS3PreSignedURLBuilder: AWSS3ObjectPresignedURLGenerator  {
+    // ...
+}
+
+extension AWSS3TransferUtility: AWSS3ObjectManager {
+    // ...
+}
+```
+
+There is an implementation in the iOS Test Suite that can be used as a reference: [aws-mobile-appsync-sdk-ios/AWSAppSyncTestCommon/S3ObjectWrapper.swift](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/blob/master/AWSAppSyncTestCommon/S3ObjectWrapper.swift). That code can be used as drop-in implementation for most use cases.
+
+The mutation operation doesn't require any specific changes in method signature. It requires only an S3ObjectInput with `bucket`, `key`, `region`, `localUri`, and `mimeType`. Now when you do a mutation, it automatically uploads the specified file to Amazon S3 using the `AWSS3TransferUtility` client internally.
 
 ## Working with Pre-Signed URLS
 
-By default, all Amazon S3 resources are private. If you want your users to have access to Amazon S3 buckets
-or objects, you can assign appropriate permissions with an [IAM policy](http://docs.aws.amazon.com/IAM/latest/UserGuide/PoliciesOverview.html).
+By default, all Amazon S3 resources are private. If you want your users to have access to Amazon S3 buckets or objects, you can assign appropriate permissions with an [IAM policy](http://docs.aws.amazon.com/IAM/latest/UserGuide/PoliciesOverview.html).
```

Comment:
Let's add a reminder of, and a link to the docs about, the fact that the Amplify CLI requires transfers to be scoped to the `public/`, `private/`, or `protected` prefixes.

Reply:
Done in the latest commit. Let me know what you think.